From fdc4a2772435806584cd03867c164e438ac19173 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 6 Dec 2021 21:23:26 -0300
Subject: [PATCH] Revert [dxso] Omit relative constant range check when
 possible

Fix tearing in some screens from Blazblue Central Fiction.
This reverts commit ce1c66d7dc404c25562f33335cec52ddb6a2c26c
---
 src/dxso/dxso_compiler.cpp | 2 ++--
 1 file changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/dxso/dxso_compiler.cpp b/src/dxso/dxso_compiler.cpp
index baf3f45..bd72de9 100644
--- a/src/dxso/dxso_compiler.cpp
+++ b/src/dxso/dxso_compiler.cpp
@@ -1029,7 +1029,7 @@ namespace dxvk {
 
       result.id = m_module.opLoad(typeId, ptrId);
 
-      if (relative && !m_moduleInfo.options.robustness2Supported) {
+      if (relative) {
         uint32_t constCount = m_module.constu32(m_layout->floatCount);
 
         // Expand condition to bvec4 since the result has four components
-- 
2.33.1

